### PR TITLE
Add auth namespace for magnificent

### LINKED ISSUE #59: Use central auth server (magnificent)
Instead of requesting data from 2-3 (closed-source, undocumented) APIs Friboo should only ask one: zalando-stups/magnificent.

### PR DESCRIPTION
**Pull request by prayerslayer**
Fixes #59:
- Add `auth` namespace
  - Contains functions `get-auth` and `require-auth` that take the request and a team and either return true/false (`get-auth`) or throw (`require-auth`).
- Switch `fetch-teams`, `fetch-service-teams` (and, as a result, `require-service-team`) to magnificent

### CODE REVIEW

**Comment by hjacobs** [comment]
I guess @dryewo and @kolja should also review :smile:

File: src/org/zalando/stups/friboo/auth.clj [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  (:require [clj-http.client :as http]
+            [org.zalando.stups.friboo.ring :as r]
+            [org.zalando.stups.friboo.config :refer [require-config]]
+            [org.zalando.stups.friboo.log :as log]
+            [io.sarnowski.swagger1st.util.api :as api]
+            [com.netflix.hystrix.core :as hystrix :refer [defcommand]]))
+
+(defcommand fetch-auth
+  "Asks magnificent if the user for this token has access to resources of this team. Returns true or false."
+  {:hystrix/fallback-fn (constantly false)}
```
Comment (hjacobs) [inline src/org/zalando/stups/friboo/auth.clj]: how does the log look like if the fallback is triggered?

**Comment by hjacobs** [comment]
No tests? :cry:

File: src/org/zalando/stups/friboo/auth.clj (hunk 1 above)
Comment (prayerslayer) [inline src/org/zalando/stups/friboo/auth.clj]: No idea, will check

Comment (prayerslayer) [inline src/org/zalando/stups/friboo/auth.clj]:
```
=> (defcommand foo {:hystrix/fallback-fn (constantly false)} (throw (new Exception "foo")))

=> (foo)
false
```

Hystrix itself doesn't log the fallback at all, apparently. I will add some.

**Comment by prayerslayer** [comment]
- Tests added
- WARN logs when fallback for `fetch-auth` is triggered

**Comment by prayerslayer** [comment]
Ping @dryewo @kolja 😃

**Comment by dryewo** [comment]
👍

**Comment by dryewo** [comment]
👍

**Pull request merged**